tests/test_other.py

```diff
@@ -9966,13 +9966,18 @@ def test_assertions_on_ready_promise(self):
     # the breaking change in #10697 Module() now returns a promise, and to get
     # the instance you must use .then() to get a callback with the instance.
     create_test_file('test.js', r'''
+      // Adds a noop catch handler to avoid unhandled promise rejection error for the process
+      function addCatch(promise) {
+        promise.catch(() => {});
+        return promise;
+      }
       try {
-        Module()._main;
+        addCatch(Module())._main;
```

Comment:
what happens without this catch? I hope the error is still shown, so for users there is no confusion - just this change was needed for testing purposes so we can continue to the next part?

if there's a way to avoid adding this, I'd be happier that way - it's better to keep tests closer to what users actually do (and I assume most don't add a catch handler, but i could be wrong).

Reply:
Correct, the error is still shown, this was just needed for testing purposes. Without a catch handler the Node process exits with a value of `7`, which I believe represents an unhandled Promise rejection, and the test fails.

Reply:
I reworked this to use the `assert_returncode` parameter instead of adding catch handlers. I didn't notice that option till just now.